src/utils/dream-lock.nix

```diff
@@ -77,6 +77,9 @@ let
 
       cyclicDependencies = lock.cyclicDependencies;
 
+      getSourceSpec = pname: version:
+        sources."${pname}"."${version}" or {};
```

Comment:
I think we should not use `{}`as a default here. If an element is accessed which doesn't exist, then it's a bug in the code and it should crash, right?

Reply:
Yeah, that makes sense. Would a warning message like in `getSource` suffice?

Reply:
Should be done in 405e63e941d6dd7ad21393a0751462363dfd215f